sources/academy/webscraping/advanced_web_scraping/crawling/crawling-sitemaps.md

```diff
@@ -0,0 +1,76 @@
+
+## How to find sitemap URLs
+
+Sitemaps are commonly restricted to contain a maximum of 50k URLs so usually, there will be a whole list of them. There can be a master sitemap containing URLs of all other sitemaps or the sitemaps might simply be indexed in robots.txt and/or have auto-incremented URLs like `/sitemap1.xml`, `/sitemap2.xml`, etc.
+
+### Google
+
+You can try your luck on Google by searching for `site:example.com sitemap.xml` or `site:example.com sitemap.xml.gz` and see if you get any results. If you do, you can try to download the sitemap and see if it contains any useful URLs. The success of this approach depends on the website telling Google to index the sitemap file itself which is rather uncommon.
+
+### robots.txt {#robots-txt}
+
+If the website has a robots.txt file, it often contains sitemap URLs. The sitemap URLs are usually listed under `Sitemap:` directive.
+
+### Common URL paths
+
+You can try to iterate over common URL paths like:
+
+- /sitemap.xml
+- /product_index.xml
+- /product_template.xml
+- /sitemap_index.xml
+- /sitemaps/sitemap_index.xml
+- /sitemap/product_index.xml
+- /media/sitemap.xml
+- /media/sitemap/sitemap.xml
+- /media/sitemap/index.xml
+
+Make also sure you test the list with `.gz`, `.tar.gz` and `.tgz` extensions and by capitalizing the words (e.g. `/Sitemap_index.xml.tar.gz`).
+
+Some websites also provide an HTML version, to help indexing bots find new content. Those include:
+
+- /sitemap
+- /category-sitemap
+- /sitemap.html
+- /sitemap_index
+
+Apify provides the [Sitemap Sniffer actor](https://apify.com/vaclavrut/sitemap-sniffer) (open-source code), that scans the URL variations automatically for you so that you don't have to check manually.
+
+## How to set up HTTP requests to download sitemaps
+
+For most sitemaps, you can make a simple HTTP request and parse the downloaded XML text with Cheerio (or just use `CheerioCrawler`). Some sitemaps are compressed and have to be streamed and decompressed. The code for that is fairly complicated so we recommend just [using Crawlee](#using-crawlee) which handles streamed and compressed sitemaps by default.
+
+## How to parse URLs from sitemaps
+
+The easiest part is to parse the actual URLs from the sitemap. The URLs are usually listed under `<loc>` tags. You can use Cheerio to parse the XML text and extract the URLs. Just be careful that the sitemap might contain other URLs that you don't want to crawl (e.g. /about, /contact, or various special category sections). [This article](/academy/node-js/scraping-from-sitemaps) provides code examples for parsing sitemaps.
+
+## Using Crawlee
+
+Fortunately, you don't have to worry about any of the above steps if you use [Crawlee](https://crawlee.dev) which has rich traversing and parsing support for sitemap. Crawlee can traverse nested sitemaps, download, and parse compressed sitemaps, and extract URLs from them. You can get all URLs in a few lines of code:
```

Comment:
```suggestion
Fortunately, you don't have to worry about any of the above steps if you use [Crawlee](https://crawlee.dev), a scraping framework, which has rich traversing and parsing support for sitemap. It can traverse nested sitemaps, download, and parse compressed sitemaps, and extract URLs from them. You can get all the URLs in a few lines of code:
```